benchmarks/benchmark_guided.py

```diff
@@ -0,0 +1,432 @@
+"""Benchmark guided decoding throughput."""
+import argparse
+import dataclasses
+import json
+import random
+import time
+from typing import List
+
+import datasets
+import uvloop
+from transformers import AutoTokenizer, PreTrainedTokenizerBase
+
+from vllm.engine.arg_utils import AsyncEngineArgs, EngineArgs
+from vllm.entrypoints.openai.api_server import (
+    build_async_engine_client_from_engine_args)
+from vllm.sampling_params import GuidedDecodingParams
+from vllm.utils import FlexibleArgumentParser, merge_async_iterators
+
+SCHEMA = {
```

Comment:
hmm, should we move this schema out to a file and we can keep a few json schema to run benchmark against.

Reply:
Move to a separate file, also added 'grammar', 'choice', 'regex' to the benchmark